src/vstest.console/CommandLine/CommandLineOptions.cs

```diff
@@ -282,17 +283,25 @@ internal Framework TargetFrameworkVersion
         /// <param name="source">Path to source file to look for tests in.</param>
         public void AddSource(string source)
         {
+            var extnList = new List<string>() { ".xap",".appx",".dll", ".exe" };
             if (String.IsNullOrWhiteSpace(source))
             {
                 throw new CommandLineException(CommandLineResources.CannotBeNullOrEmpty);
             }
             
             source = source.Trim();
-            
+
             // Convert the relative path to absolute path
             if (!Path.IsPathRooted(source))
             {
+                if (!extnList.Contains(source))
```

Comment:
>source [](start = 39, length = 6)

this looks wrong, it will always fail in case path is not rooted.

Reply:
Sorry, I have updated logic and works when it is not rooted(relative path)